ctapipe/core/container.py

```diff
@@ -37,72 +91,52 @@ class Container:
     >>>    cont = MyContainer()
     >>>    print(cont.x)
     100
-    >>>    # metdata will become header keywords in an output file:
-    >>>    cont.meta['KEY'] = value  
+    >>>    # metadata will become header keywords in an output file:
+    >>>    cont.meta['KEY'] = value
 
     `Items` inside `Containers` can contain instances of other
     `Containers`, to allow for a hierarchy of containers, and can also
     contain a `Map` for the case where one wants e.g. a set of
     sub-classes indexed by a value like the `telescope_id`. Examples
     of this can be found in `ctapipe.io.containers`
 
-    `Containers` work by shadowing all class variables (which must be
-    instances of `Item`) with instance variables of the same name the
-    hold the value expected. If `Container.reset()` is called, all
-    instance variables are reset to their default values as defined in
-    the class.
-
     Finally, `Containers` can have associated metadata via their
     `meta` attribute, which is a `dict` of keywords to values.
 
     """
 
-    def __init__(self, **values):
-        object.__setattr__(self, "_metadata", dict())
-        self.reset()
-        for key in values:
-            self[key] = values[key]
+    def __init__(self, **items):
 
-    def __setattr__(self, name, value):
-        """Prevent new attributes that aren't in the class definition"""
-        if hasattr(self.__class__, name):
-            object.__setattr__(self, name, value)
-        else:
-            raise AttributeError(
-                "{} has no attribute '{}'".format(self.__class__, name))
+        self.meta = {}
+        for k, v in self._items.items():
+            setattr(self, k, deepcopy(v.default))
+            self.meta[k] = v
 
-    def __getitem__(self, item):
-        return self.__dict__[item]
+        for k, v in items.items():
+            setattr(self, k, v)
 
-    def __setitem__(self, key, value):
-        if hasattr(self.__class__, key):
-            self.__dict__[key] = value
+    def __getitem__(self, item):
+        if item in self.__slots__:
+            return getattr(self, item)
         else:
-            raise AttributeError("{} has no attribute '{}'"
-                .format(self.__class__,key))
-
-    @property
-    def meta(self):
-        """metadata key/values associated with this Container.
+            raise KeyError('No such Item: "{}"'.format(item))
 
-        When written to an output file, these will become headers, so
-        should represent data that does not change after the
-        `Container` is constructed.
-        """
-        return self._metadata
+    def __setitem__(self, item, value):
```

Comment:
This is to be able to use bracket access. But your right, the check shouldnt be necessary.